app/__init__.py

```diff
@@ -0,0 +1,71 @@
+import base64
+import hashlib
+import logging
+
+from django.conf import settings
+from django.contrib.auth import authenticate, login
+from django.contrib.auth.models import User
+from django.core.urlresolvers import reverse
+from django.db import IntegrityError
+
+from django_lti_tool_provider import AbstractApplicationHookManager
+from django_lti_tool_provider.views import LTIView
+
+
+logger = logging.getLogger(__name__)
+
+
+class ApplicationHookManager(AbstractApplicationHookManager):
+    LTI_KEYS = ['']
+
+    @classmethod
+    def _compress_user_name(cls, username):
+        try:
+            binary = username.decode('hex')
```

Comment:
@itsjeyd Is there a test that covers this particular case?

Reply:
@tomaszgy Yup, [`test_authentication_hook_user_exists`](https://github.com/open-craft/learner-profile-dashboard/blob/basic-implementation/app/tests/test_auth.py#L31) is set up to run with an [item](https://github.com/open-craft/learner-profile-dashboard/blob/basic-implementation/app/tests/test_auth.py#L29) that contains a `user_id` for which `.decode('hex')` does not raise a `TypeError`:

```
In [1]: 'abcdef1234567890'.decode('hex')
Out[1]: '\xab\xcd\xef\x124Vx\x90'
```